hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSplit.java

```diff
@@ -184,9 +184,24 @@ public TableSplit(final byte [] tableName, byte[] startRow, byte[] endRow,
    * @param startRow The start row of the split.
    * @param endRow The end row of the split.
    * @param location The location of the region.
+   * @param encodedRegionName The region ID.
+   * @param length Size of region in bytes
    */
   public TableSplit(TableName tableName, byte[] startRow, byte[] endRow,
-      final String location) {
+    final String location, final String encodedRegionName, long length) {
+    this(tableName, null, startRow, endRow, location, encodedRegionName, length);
```

Comment:
Why do we need a new constructor? Just pass null wherever it is not needed?

Reply:
@bharathv I think the separate constructor is more intuitive that it is okay to not have scan object? What do you think?

Reply:
@Reidddddd Scan is not useless for all the input formats of map-reduce, it is for the `TableInputFormat`.
We do need the scan object for `MultiTableInputFormat`